### DISCUSSION TITLE
0.9.0-stable has been released!

### DISCUSSION THREAD

**Discussion by mihe** [godot-jolt/godot-jolt#641] [Announcements]
This release supports [**Godot 4.1**](https://github.com/godotengine/godot/releases/tag/4.1.2-stable).

## Changelog

These are the notable changes that have been made since [0.8.0-stable](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.8.0-stable) was released. You can also find a list of all the commits [here](https://github.com/godot-jolt/godot-jolt/compare/v0.8.0-stable...v0.9.0-stable).

Any breaking changes are denoted with ⚠️.

### Changed

- Changed `ConvexPolygonShape3D` to no longer emit errors about failing to build the shape when adding one to the scene tree with 0 points.

### Added

- Added new project setting, "Active Edge Threshold", for tuning the cut-off angle for Jolt's active edge detection, which can help balance trade-offs related to triangle edge collisions.

### Fixed

- ⚠️ Fixed issue where `Generic6DOFJoint` and `JoltGeneric6DOFJoint` would lock up any axis that used a spring stiffness/frequency of 0.
- Greatly reduced creation/modification/loading times for `ConcavePolygonShape3D`.

<hr /><em>This discussion was created from the release <a href='https://github.com/godot-jolt/godot-jolt/releases/tag/v0.9.0-stable'>0.9.0-stable</a>.</em>

**Comment by ywmaa** [comment]
Thank you for your awesome efforts, this release allows me to use jolt in Godot 4.2 beta 1 with concave collision shapes.

**Comment by Gamepro5** [comment]
> Added new project setting, "Active Edge Threshold", for tuning the cut-off angle for Jolt's active edge detection, which can help balance trade-offs related to triangle edge collisions.

Thank you!!

**Comment by Conz3D** [comment]
Has there something changed with "Debug" -> "Visible Collision Shapes" in 0.9?
Some contact points (the bright, red triangles) seem to be offset in some cases. Esp. with compound objects and shapes attached with joints.
Haven't seen this effect in 0,8. It looks like it's only the debug display, and the behavior of the objects seems ok. Is this a known issue?

![jolt_debug_display_bug01](https://github.com/godot-jolt/godot-jolt/assets/129608680/f2b7abb1-3fce-493b-9fb8-d0b401c1289b)

**Reply by Gamepro5** [reply]
It looks like your speculative distance is set to very high. Can you make an MRP?

**Reply by Conz3D** [reply]
It already got fixed and merged by mihe. It was displaying contact points between RigidBody3D and Area3D.
https://github.com/godot-jolt/godot-jolt/pull/651
(What is an MRP?)

**Reply by Gamepro5** [reply]
minimal reproduction project. It helps debugging. I thought it was speculative distance because when i set it super high it looked like that with visible collision shapes.
